vernac/indschemes.ml

```diff
@@ -321,121 +313,97 @@ let declare_sym_scheme ind =
 
 (* Scheme command *)
 
-let smart_global_inductive y = smart_global_inductive y
-let rec split_scheme env l =
+(* Boolean on scheme_type cheking if it considered dependent *)
+let sch_isdep = function
+| SchemeInduction  | SchemeElimination -> true
+| SchemeMinimality | SchemeCase        -> false
+
+(* Generate suffix for scheme given a target sort *)
+let scheme_suffix_gen {sch_type; sch_qualid; sch_sort} sort =
+  (* We check if we are working with recursion vs case schemes *)
+  let sch_isrec = match sch_type with
+    | SchemeInduction   | SchemeMinimality  -> true
+    | SchemeElimination | SchemeCase        -> false in
+  (* The _ind/_rec_/case suffix *)
+  let ind_suffix = match sch_isrec , sch_sort with
+    | true  , InSProp
+    | true  , InProp  -> "_ind"
+    | true  , _       -> "_rec"
+    | false , _       -> "_case" in
+  (* SProp and Type have an auxillary ending to the _ind suffix *)
+  let aux_suffix = match sch_sort with
+    | InSProp -> "s"
+    | InType  -> "t"
+    | _       -> "" in
+  (* Some schemes are deliminated with _dep or no_dep *)
+  let dep_suffix = match sch_isdep sch_type , sort with
+    | true  , InProp  -> "_dep"
+    | false , InSet
+    | false , InType
+    | false , InSProp -> "_nodep"
+    | _ , _           -> "" in
+  ind_suffix ^ aux_suffix ^ dep_suffix
```

Comment:
@jfehrle Here is how the naming works. As you can see it is a bit tricky to describe succinctly in the doc. This code comes from me explicitly trying all the combinations and writing this nice little suffix generator explicitly. Before it was a real mess in the code. I'm not sure what use it will be to document the exact format of the names as they can change a bit arbitrarily. When a scheme command is run, the user will see the message `Foo_ind has been defined` and also they have the option of defining their own name. I therefore don't think it is worth going into too much detail.

Reply:
> I therefore don't think it is worth going into too much detail.

Agreed.  Vagueness is sometime a virtue.